Turkey/TestRunner.cs

```diff
@@ -44,10 +44,12 @@ public class SystemUnderTest
 
     public class TestOutput
     {
+        public async virtual Task BeforeTesting() {}
```

Comment:
Instead of adding more to this class, may be we should refactor how these pieces fit together.

How about:
- moving the `Console` output that is generated by `NewOutput` directly into the `TestRunner` class, and remove `NewOutput`.
- adding a Class that describes a test result completely, something like:
```cs
class ExecutedTest {
    public string Name { get; init; }
    public TestResult Result { get; init; }
    public StringBuilder Log { get; init; }
}
```
- Let the `TestRunner` track these in a `List<ExecutedTest>`.
- Rename `TestOutput` to `TestReport` and have a single method `WriteResults` that accepts this list as an argument, and update JUnitOutput accordingly.

cc @omajid

Reply:
I am generally against this proposal.

We (Radka and I) designed this class to make the output formats pluggable/changeable through different implementations. We even had different implementations based on different environments where tests were running.  Moving the console output into `TestRunner` means we get locked into UX, without different pluggable implementations. Moving everything via a single `WriteResults` means that the report/UI can only be generated/updated once all tests are done, and not as tests are running.